test/mapmatch.cc

```diff
@@ -167,27 +169,44 @@ void test_matcher() {
     }
     // get the edges along that route shape
     boost::property_tree::ptree walked;
+    std::string walked_json;
     try {
-      walked = json_to_pt(actor.trace_attributes(
+      walked_json = actor.trace_attributes(
           R"({"costing":"auto","shape_match":"edge_walk","encoded_polyline":")" +
-          json_escape(encoded_shape) + "\"}"));
+          json_escape(encoded_shape) + "\"}");
+      walked = json_to_pt(walked_json);
     } catch (...) {
       std::cout << test_case << std::endl;
       std::cout << R"({"costing":"auto","shape_match":"edge_walk","encoded_polyline":")" +
                        json_escape(encoded_shape) + "\"}"
                 << std::endl;
       throw std::logic_error("Edge walk failed with exact shape");
     }
+
+    // check the shape makes sense
+    auto walked_encoded_shape = walked.get<std::string>("shape");
+    auto walked_shape = midgard::decode<std::vector<midgard::PointLL>>(walked_encoded_shape);
+    /*if (walked_shape.size() != shape.size()) {
+      throw std::logic_error("Differing shape lengths " + std::to_string(shape.size()) +
```

Comment:
once i had made the change to do the trimming properly, this test started failing with some wild numbers in terms of distance between points. so i eventually found out that when you take a route and you then edge_walk map match its shape, the shape that you get back has 1 more point in it. i added this code to determine that. even after reverting all my o ther changes but leaving this check, the code still breaks. so this commented code is another bug to fix unrelated to this pr about why the shape walk of a regular route has one more shape point.